internal/checker/printer.go

```diff
@@ -188,7 +188,13 @@ func (p *Printer) printTypeNoAlias(t *Type) {
 	case t.flags&TypeFlagsStringMapping != 0:
 		p.printStringMappingType(t)
 	case t.flags&TypeFlagsSubstitution != 0:
-		p.printType(t.AsSubstitutionType().baseType)
+		noInferSymbol := core.IfElse(p.c.isNoInferType(t), p.c.getGlobalNoInferSymbolOrNil(), nil)
+		if noInferSymbol != nil {
```

Comment:
I wouldn't use an `IfElse` here; an if statement suffices and will be faster since it'll avoid calling `getGlobalNoInferSymbolOrNil` in the common case.

Reply:
done